Ginger/DatabaseUnitTest/DataBaseUnitTest.cs

```diff
@@ -0,0 +1,151 @@
+        public static void ClassInit(TestContext context)
+        {
+            Dictionary<string, string> param = new Dictionary<string, string>();
+            string FilePath = TestResources.GetTestResourcesFile(@"SignUp.accdb");
+            param.Add("ConnectionString", @"Provider=Microsoft.ACE.OLEDB.12.0;Data Source=" + FilePath +";");
+            db.KeyvalParamatersList = param;
+            Boolean testconn = db.OpenConnection(param);
+            
+        }
+
+        [TestMethod]
+        public void OpenConnection()
+        {
+            //Arrange
+            Dictionary<string, string> param = new Dictionary<string, string>();
+            param.Add("ConnectionString", @"Provider=Microsoft.ACE.OLEDB.12.0;Data Source= C:\GingerSourceControl\Solutions\Ginger_Regression_Testing\Documents\WOEI\SignUp.accdb;");
+
+            //Act
+            Boolean testconn = db.OpenConnection(param);
+
+            //Assert
+            Assert.AreEqual(testconn, true);
+        }
+
+        [TestMethod]
+        public void GetTableList()
+        {
+            //Arrange
+            
+            List<string> Tables= null;
+            
+            //Act
+
+             Tables= db.GetTablesList();
+           
+            //Assert
+            Assert.AreEqual(Tables.Count, 2);
+        }
+
+        [TestMethod]
+        public void GetTablesColumns()
+        {
+            //Arrange
+            List<string> Columns = null;
+            string tablename = "Person";
+            
+            //Act
+            Columns = db.GetTablesColumns(tablename);
+           
+            //Assert
+            Assert.AreEqual(Columns.Count, 9);
+        }
+
+        [TestMethod]
+        public void RunUpdateCommand()
+        {
+            //Arrange
+            string upadateCommand = "update Person set LName=\"EFGH\" where ID=1";
+            string result = null;
+            
+            //Act
+            result = db.RunUpdateCommand(upadateCommand, false);
+           
+            //Assert
+            Assert.AreEqual(result, "1");
+        }
+
+        [TestMethod]
+        public void GetSingleValue()
+        {
+            //Arrange
+            string result = null;
+            
+            //Act
+            result = db.GetSingleValue("Person", "FName", "ID=2");
+            
+            //Assert
+            Assert.AreEqual(result, "LMPO");
+        }
+
+        [TestMethod]
+        public void DBQuery()
+        {
+            //Arrange
+            List<object> result = null;
+            
+            //Act
+            result = db.DBQuery("Select * from Person");
+            
+            //Assert
+            Assert.AreEqual(result.Count, 2);
+        }
+
+        [TestMethod]
+        public void GetRecordCount()
+        {
+            //Arrange
+             int a = 0;
+            
+            //Act
+                a = db.GetRecordCount("Person");
+            
+            //Assert
+            Assert.AreEqual(a, 2);
```

Comment:
2 is expected needs to be first param, then add ex info what we check so if it fails the err will be clear

Reply:
please handle

Reply:
added check